In [0]:
var2 = 5
print('Var is greater than 10') if var2 >10 else print('Var2 is less than 10')
  

Var2 is less than 10

In [0]:
Type casting 

In [0]:
help(range)

In [0]:
# string,
string_type = 'single quote'
type(string_type)
string_type="double quote"
type(string_type)

Out[4]: str

#### Data Types...
* 1) STR
* 2) INT
* 3) FLOAT
* 4) COMPLEX
* 5) LIST
* 6) TUPLE
* 7) RANGE
* 8) DICT
* 9) SET
* 10) FROZENSET
* 11) BOOL
* 12) BYTES
* 13) BYTEARRAY
* 14) MEMORYVIEW

##### Range data Type
* The range() function is used to generate a sequence of numbers over time. 
* At its simplest, it accepts an integer and returns a range object (a type of iterable)
* syntax: `range(start, stop, step)`
* `start`	Optional. An integer number specifying at which position to start. Default is 0
* `stop`	Required. An integer number specifying at which position to stop (not included).
* `step`	Optional. An integer number specifying the incrementation. Default is 1

In [0]:
v_range = range(10,100,5) # n-1


In [0]:
list(v_range)

Out[31]: [10, 15, 20, 25, 30, 35, 40, 45, 50, 55, 60, 65, 70, 75, 80, 85, 90, 95]

In [0]:
v_range =range(5,100,5)


5
10
15
20
25
30
35
40
45
50
55
60
65
70
75
80
85
90
95

In [0]:
range_v = range(10)
print(type(range_v))
print(tuple(range_v))

<class 'range'>
(0, 1, 2, 3, 4, 5, 6, 7, 8, 9)

In [0]:
range_v=range(10)
print(range_v)
print(list(range_v))

range(0, 10)
[0, 1, 2, 3, 4, 5, 6, 7, 8, 9]


In [0]:
v_range = range(3,10)
print('Range values are generated : ',list(v_range))
print('v_range data type is : ', type(v_range))

Range values are generated : [3, 4, 5, 6, 7, 8, 9]
v_range data type is : <class 'range'>

In [0]:
list(range(-100, -95))

Out[5]: [-100, -99, -98, -97, -96]

In [0]:
list(range(1, 20, 5))  # Increment by 5

Out[7]: [1, 6, 11, 16]

In [0]:
x_range = range(50,150)
print(list(x_range))
print(type(x_range))

[50, 51, 52, 53, 54, 55, 56, 57, 58, 59, 60, 61, 62, 63, 64, 65, 66, 67, 68, 69, 70, 71, 72, 73, 74, 75, 76, 77, 78, 79, 80, 81, 82, 83, 84, 85, 86, 87, 88, 89, 90, 91, 92, 93, 94, 95, 96, 97, 98, 99, 100, 101, 102, 103, 104, 105, 106, 107, 108, 109, 110, 111, 112, 113, 114, 115, 116, 117, 118, 119, 120, 121, 122, 123, 124, 125, 126, 127, 128, 129, 130, 131, 132, 133, 134, 135, 136, 137, 138, 139, 140, 141, 142, 143, 144, 145, 146, 147, 148, 149]
<class 'range'>


In [0]:
range_x=range(100,120)
print(list(range_x))

In [0]:
x = range(6)	#range	
print('Range Data Type : ',type(x))
print(x)
print(list(x))
y = range(1,10)
print(list(y))

#### String Data Type
* Strings are amongst the most popular types in Python. We can create them simply by enclosing characters in quotes.
* Python treats single quotes the same as double quotes. 
* Creating strings is as simple as assigning a value to a variable

In [0]:
help(x)

In [0]:
y_str = 'This is pyspark Telugu Youtube channel'
x = "Hello World"	#str
print('String Data Type: ',type(x))
print(x)
print(y_str)
print(type(y_str))

String Data Type: <class 'str'>
Hello World
This is pyspark Telugu Youtube channel
<class 'str'>

#### Int Data Type
* hey are positive or negative whole numbers with no decimal point. Integers in Python 3 are of unlimited size.

In [0]:
a=55
a
type(a)

Out[11]: int

In [0]:
x = 34	#int	
y = 534 #int
print('Int Data Type : ',type(x))
print(x)
print(type(y))

Int Data Type : <class 'int'>
34
<class 'int'>

#### Float Data Type
* they represent real numbers and are written with a decimal point dividing the integer and the fractional parts. 
* Floats may also be in scientific notation, with E or e indicating the power of 10 (2.5e2 = 2.5 x 102 = 250).

In [0]:
x = 20.2342	#float	
print('Float Data Type : ',type(x))
print(x)

Float Data Type : <class 'float'>
20.2342

#### List
* List is a collection which is ordered and changeable. Allows duplicate members.
* In Python lists are written with square brackets.

In [0]:
bool_v =[1,2,3,"sfsdf",True,'sdfsd']
type(bool_v)

Out[17]: list

In [0]:
a = [1,0,"this is index 2",1,2,2,2,3,4,5,6,7,8,'String','NAme',44.33,True]
print(type(a))
print(a[2])

<class 'list'>
this is index 2

In [0]:
x = ["apple",23,2323,34.34, "banana", "cherry",11,22,11,22,1,1,2,2,2,3,3]	#list	
print(' List Data Type : ',type(x))
print(x[2])

List Data Type : <class 'list'>
2323

In [0]:
num_list=[1,2,3,4,5,6,7,7,7,6,6,5,5,8,8,9,9,]
print(type(num_list))
print(num_list)

#### Tuple
* Tuple is a collection which is ordered and unchangeable. Allows duplicate members.
* In Python tuples are written with round brackets.



In [0]:
a_tuple=(1,2,3,4)
type(a_tuple)

Out[19]: tuple

In [0]:
x = ("apple", 12,1212,121.12,"banana", "cherry")	#tuple	
print('Tuple Data Type : ', type(x))
print(x[0])

Tuple Data Type : <class 'tuple'>
apple

In [0]:
num_tuple = (1,2,3,4,5,6,6,7,7,8,8,9,9,10)
print(type(num_tuple))
print(num_tuple)

#### Dictionary
* Dictionary is a collection which is unordered, changeable and indexed. No duplicate members.

* In Python dictionaries are written with curly brackets, and they have keys and values.



In [0]:
x = {"name" : "John", "age" : 36,"loc":"Bangalore"}	#dict	
print('Dictionary Data Type : ',type(x))
print(x['loc'])

Dictionary Data Type : <class 'dict'>
Bangalore

In [0]:
a_set = set()


#### Sets
* A set is a collection which is unordered and unindexed.  No duplicate members.

* In Python, sets are written with curly brackets.



In [0]:
x = {"apple", "banana", "cherry","cherry"}	#set	
print('Set Data Type : ',type(x))
print(x)

Set Data Type : <class 'set'>
{'apple', 'cherry', 'banana'}

In [0]:
a_set = {1,1,1,2,2,2,3,4,4,5,5,5,6,7,8,9}
a_set

Out[8]: {1, 2, 3, 4, 5, 6, 7, 8, 9}

#### frozenset()
* The frozenset() is an inbuilt function is Python which takes an iterable object as input and makes them immutable. Simply it freezes the iterable objects and makes them unchangeable.


In [0]:
x = frozenset({"apple", "banana", "cherry"})	#frozenset	
print('Frozenset Data Type : ',type(x))
print(x)

Frozenset Data Type : <class 'frozenset'>
frozenset({'apple', 'cherry', 'banana'})

#### Bool
* The boolean data type is either True or False. In Python, boolean variables are defined by the True and False keywords.

In [0]:
a=False
b=True
print('variable a type is : ',type(a))
print('variable b type is : ',type(b))

variable a type is : <class 'bool'>
variable b type is : <class 'bool'>

In [0]:
num =0
bool(num)
# y or n  or True or False or 1 or 0 or active or inactive

Out[37]: False

In [0]:
x = False	#bool	
print('Bool Data Type : ',type(x))
print(x)

In [0]:
bool_f=False
print(type(bool_f))
print(bool_f)

#### Bytes
* Return a new "bytes" object, which is an immutable sequence of small integers in the range 0 <= x < 256, print as ASCII characters when displayed. bytes is an immutable version of bytearray – it has the same non-mutating methods and the same indexing and slicing behavior.

In [0]:
x = b"Hello"	#bytes	
print('Bytes Data Type : ',type(x))
print(x)

Bytes Data Type : <class 'bytes'>
b'Hello'

In [0]:
type(x)

Out[35]: bytes

#### bytearray()
* bytearray() method returns a bytearray object which is an array of given bytes. It gives a mutable sequence of integers in the range 0 <= x < 256.
* The difference between bytes() and bytearray() is that bytes() returns an object that cannot be modified, and bytearray() returns an object that can be modified.


In [0]:
# Converting inter into bytearray
x = bytearray([14]) 	#bytearray	
print('Byte Array Data Type : ',type(x))
print(x)

Byte Array Data Type : <class 'bytearray'>
bytearray(b'\x0e')

#### Memoryview()
* The memoryview() function returns a memory view object from a specified object.
* A memory view is a safe way to expose the buffer protocol in Python.
* It allows you to access the internal buffers of an object by creating a memory view object.

In [0]:
x = memoryview(bytes(5))	#memoryview
print('Memory View Data Type : ',type(x))
x = memoryview(b"Hello")
print(x)
#return the Unicode of the first character
print(x[0])
#return the Unicode of the second character
print(x[1])

Memory View Data Type : <class 'memoryview'>
<memory at 0x7f47985907c0>
72
101

In [0]:
type(x)

Out[43]: memoryview

#### Complex integer data type
* Complex numbers have a real and imaginary part, which are each implemented using double in C. 
* To extract these parts from a complex number z, use z.real and z.imag.


* Returns a complex number constructed from arguments
* Complex numbers are specified as `<real part>+<imaginary part>j`.  
* J (or j) represents the square root of -1 (which is an imaginary number)
* Syntax `complex(re,im) ` a complex number with real part re, imaginary part im. im defaults to zero.	
* Complex numbers have a real and imaginary part, which are each implemented using double in C. To extract these parts from a complex number z, use z.real and z.imag.

In [0]:
x = 10+55j	#complex	(method or function )
print('Complex Data Type : ',type(x))
print(x) 
print('real value  : ' ,x.real)
print('image value : ',x.imag)

Complex Data Type : <class 'complex'>
(10+55j)
real value : 10.0
image value : 55.0

In [0]:
a = 6+4j
b = 3+2j
print("Below are the various functions we can use with complex numbers:")
print("1->Addition of two complex numbers:",a+b)
print("2->Addition of two complex numbers:",a-b)
print("3->Addition of two complex numbers:",a*b)
print("4->Addition of two complex numbers:",a/b)
#print("5->Addition of two complex numbers:",a//b) -->TypeError: can't take floor of complex number.
#print("6->Addition of two complex numbers:",a%b)--> TypeError: can't mod complex numbers.

Below are the various functions we can use with complex numbers:
1->Addition of two complex numbers: (9+6j)
2->Addition of two complex numbers: (3+2j)
3->Addition of two complex numbers: (10+24j)
4->Addition of two complex numbers: (2+0j)


####  Date & Time Types


In [0]:
import datetime
x = datetime.datetime.now()
print(x)
type(x)

2021-05-05 04:10:42.049061
Out[44]: datetime.datetime

In [0]:
help(x)

Help on datetime object:

class datetime(date)
 datetime(year, month, day[, hour[, minute[, second[, microsecond[,tzinfo]]]]])
 
 The year, month and day arguments are required. tzinfo may be None, or an
 instance of a tzinfo subclass. The remaining arguments may be ints.
 
 Method resolution order:
 datetime
 date
 builtins.object
 
 Methods defined here:
 
 __add__(self, value, /)
 Return self+value.
 
 __eq__(self, value, /)
 Return self==value.
 
 __ge__(self, value, /)
 Return self>=value.
 
 __getattribute__(self, name, /)
 Return getattr(self, name).
 
 __gt__(self, value, /)
 Return self>value.
 
 __hash__(self, /)
 Return hash(self).
 
 __le__(self, value, /)
 Return self<=value.
 
 __lt__(self, value, /)
 Return self<value.
 
 __ne__(self, value, /)
 Return self!=value.
 
 __radd__(self, value, /)
 Return value+self.
 
 __reduce__(...)
 __reduce__() -> (cls, state)
 
 __reduce_ex__(...)
 __reduce_ex__(proto) -> (cls, state)
 
 __repr__(self, /)
 Return repr(self).
 
 __rsub__(self, value, /)
 Return value-self.
 
 __str__(self, /)
 Return str(self).
 
 __sub__(self, value, /)
 Return self-value.
 
 astimezone(...)
 tz -> convert to local time in new timezone tz
 
 ctime(...)
 Return ctime() style string.
 
 date(...)
 Return date object with same year, month and day.
 
 dst(...)
 Return self.tzinfo.dst(self).
 
 isoformat(...)
 [sep] -> string in ISO 8601 format, YYYY-MM-DDT[HH[:MM[:SS[.mmm[uuu]]]]][+HH:MM].
 sep is used to separate the year from the time, and defaults to 'T'.
 timespec specifies what components of the time to include (allowed values are 'auto', 'hours', 'minutes', 'seconds', 'milliseconds', and 'microseconds').
 
 replace(...)
 Return datetime with new specified fields.
 
 time(...)
 Return time object with same time but with tzinfo=None.
 
 timestamp(...)
 Return POSIX timestamp as float.
 
 timetuple(...)
 Return time tuple, compatible with time.localtime().
 
 timetz(...)
 Return time object with same time and tzinfo.
 
 tzname(...)
 Return self.tzinfo.tzname(self).
 
 utcoffset(...)
 Return self.tzinfo.utcoffset(self).
 
 utctimetuple(...)
 Return UTC time tuple, compatible with time.localtime().
 
 ----------------------------------------------------------------------
 Class methods defined here:
 
 combine(...) from builtins.type
 date, time -> datetime with same date and time fields
 
 fromisoformat(...) from builtins.type
 string -> datetime from datetime.isoformat() output
 
 fromtimestamp(...) from builtins.type
 timestamp[, tz] -> tz's local time from POSIX timestamp.
 
 now(tz=None) from builtins.type
 Returns new datetime object representing current time local to tz.
 
 tz
 Timezone object.
 
 If no tz is specified, uses local timezone.
 
 strptime(...) from builtins.type
 string, format -> new datetime parsed from a string (like time.strptime()).
 
 utcfromtimestamp(...) from builtins.type
 Construct a naive UTC datetime from a POSIX timestamp.
 
 utcnow(...) from builtins.type
 Return a new datetime representing UTC day and time.
 
 ----------------------------------------------------------------------
 Static methods defined here:
 
 __new__(*args, **kwargs) from builtins.type
 Create and return a new object. See help(type) for accurate signature.
 
 ----------------------------------------------------------------------
 Data descriptors defined here:
 
 fold
 
 hour
 
 microsecond
 
 minute
 
 second
 
 tzinfo
 
 ----------------------------------------------------------------------
 Data and other attributes defined here:
 
 max = datetime.datetime(9999, 12, 31, 23, 59, 59, 999999)
 
 min = datetime.datetime(1, 1, 1, 0, 0)
 
 resolution = datetime.timedelta(microseconds=1)
 
 ----------------------------------------------------------------------
 Methods inherited from date:
 
 __format__(...)
 Formats self with strftime.
 
 isocalendar(...)
 Return a 3-tuple containing ISO year, week number, and weekday.
 
 isoweekday(...)
 Return the day of the week represented by the date.
 Mond

#### strftime()
* The method is called strftime(), and takes one parameter, format, to specify the format of the returned string:

* Options

%a	Weekday, short version	Wed	
%A	Weekday, full version	Wednesday	
%w	Weekday as a number 0-6, 0 is Sunday	3	
%d	Day of month 01-31	31	
%b	Month name, short version	Dec	
%B	Month name, full version	December	
%m	Month as a number 01-12	12	
%y	Year, short version, without century	18	
%Y	Year, full version	2018	
%H	Hour 00-23	17	
%I	Hour 00-12	05	
%p	AM/PM	PM	
%M	Minute 00-59	41	
%S	Second 00-59	08	
%f	Microsecond 000000-999999	548513	
%z	UTC offset	+0100	
%Z	Timezone	CST	
%j	Day number of year 001-366	365	
%U	Week number of year, Sunday as the first day of week, 00-53	52	
%W	Week number of year, Monday as the first day of week, 00-53	52	
%c	Local version of date and time	Mon Dec 31 17:41:00 2018	
%x	Local version of date	12/31/18	
%X	Local version of time	17:41:00	
%%	A % character	%

In [0]:
help(today_date)

Help on datetime object:

class datetime(date)
 datetime(year, month, day[, hour[, minute[, second[, microsecond[,tzinfo]]]]])
 
 The year, month and day arguments are required. tzinfo may be None, or an
 instance of a tzinfo subclass. The remaining arguments may be ints.
 
 Method resolution order:
 datetime
 date
 builtins.object
 
 Methods defined here:
 
 __add__(self, value, /)
 Return self+value.
 
 __eq__(self, value, /)
 Return self==value.
 
 __ge__(self, value, /)
 Return self>=value.
 
 __getattribute__(self, name, /)
 Return getattr(self, name).
 
 __gt__(self, value, /)
 Return self>value.
 
 __hash__(self, /)
 Return hash(self).
 
 __le__(self, value, /)
 Return self<=value.
 
 __lt__(self, value, /)
 Return self<value.
 
 __ne__(self, value, /)
 Return self!=value.
 
 __radd__(self, value, /)
 Return value+self.
 
 __reduce__(...)
 __reduce__() -> (cls, state)
 
 __reduce_ex__(...)
 __reduce_ex__(proto) -> (cls, state)
 
 __repr__(self, /)
 Return repr(self).
 
 __rsub__(self, value, /)
 Return value-self.
 
 __str__(self, /)
 Return str(self).
 
 __sub__(self, value, /)
 Return self-value.
 
 astimezone(...)
 tz -> convert to local time in new timezone tz
 
 ctime(...)
 Return ctime() style string.
 
 date(...)
 Return date object with same year, month and day.
 
 dst(...)
 Return self.tzinfo.dst(self).
 
 isoformat(...)
 [sep] -> string in ISO 8601 format, YYYY-MM-DDT[HH[:MM[:SS[.mmm[uuu]]]]][+HH:MM].
 sep is used to separate the year from the time, and defaults to 'T'.
 timespec specifies what components of the time to include (allowed values are 'auto', 'hours', 'minutes', 'seconds', 'milliseconds', and 'microseconds').
 
 replace(...)
 Return datetime with new specified fields.
 
 time(...)
 Return time object with same time but with tzinfo=None.
 
 timestamp(...)
 Return POSIX timestamp as float.
 
 timetuple(...)
 Return time tuple, compatible with time.localtime().
 
 timetz(...)
 Return time object with same time and tzinfo.
 
 tzname(...)
 Return self.tzinfo.tzname(self).
 
 utcoffset(...)
 Return self.tzinfo.utcoffset(self).
 
 utctimetuple(...)
 Return UTC time tuple, compatible with time.localtime().
 
 ----------------------------------------------------------------------
 Class methods defined here:
 
 combine(...) from builtins.type
 date, time -> datetime with same date and time fields
 
 fromisoformat(...) from builtins.type
 string -> datetime from datetime.isoformat() output
 
 fromtimestamp(...) from builtins.type
 timestamp[, tz] -> tz's local time from POSIX timestamp.
 
 now(tz=None) from builtins.type
 Returns new datetime object representing current time local to tz.
 
 tz
 Timezone object.
 
 If no tz is specified, uses local timezone.
 
 strptime(...) from builtins.type
 string, format -> new datetime parsed from a string (like time.strptime()).
 
 utcfromtimestamp(...) from builtins.type
 Construct a naive UTC datetime from a POSIX timestamp.
 
 utcnow(...) from builtins.type
 Return a new datetime representing UTC day and time.
 
 ----------------------------------------------------------------------
 Static methods defined here:
 
 __new__(*args, **kwargs) from builtins.type
 Create and return a new object. See help(type) for accurate signature.
 
 ----------------------------------------------------------------------
 Data descriptors defined here:
 
 fold
 
 hour
 
 microsecond
 
 minute
 
 second
 
 tzinfo
 
 ----------------------------------------------------------------------
 Data and other attributes defined here:
 
 max = datetime.datetime(9999, 12, 31, 23, 59, 59, 999999)
 
 min = datetime.datetime(1, 1, 1, 0, 0)
 
 resolution = datetime.timedelta(microseconds=1)
 
 ----------------------------------------------------------------------
 Methods inherited from date:
 
 __format__(...)
 Formats self with strftime.
 
 isocalendar(...)
 Return a 3-tuple containing ISO year, week number, and weekday.
 
 isoweekday(...)
 Return the day of the week represented by the date.
 Mond

In [0]:
help(today_date)

Help on datetime object:

class datetime(date)
 datetime(year, month, day[, hour[, minute[, second[, microsecond[,tzinfo]]]]])
 
 The year, month and day arguments are required. tzinfo may be None, or an
 instance of a tzinfo subclass. The remaining arguments may be ints.
 
 Method resolution order:
 datetime
 date
 builtins.object
 
 Methods defined here:
 
 __add__(self, value, /)
 Return self+value.
 
 __eq__(self, value, /)
 Return self==value.
 
 __ge__(self, value, /)
 Return self>=value.
 
 __getattribute__(self, name, /)
 Return getattr(self, name).
 
 __gt__(self, value, /)
 Return self>value.
 
 __hash__(self, /)
 Return hash(self).
 
 __le__(self, value, /)
 Return self<=value.
 
 __lt__(self, value, /)
 Return self<value.
 
 __ne__(self, value, /)
 Return self!=value.
 
 __radd__(self, value, /)
 Return value+self.
 
 __reduce__(...)
 __reduce__() -> (cls, state)
 
 __reduce_ex__(...)
 __reduce_ex__(proto) -> (cls, state)
 
 __repr__(self, /)
 Return repr(self).
 
 __rsub__(self, value, /)
 Return value-self.
 
 __str__(self, /)
 Return str(self).
 
 __sub__(self, value, /)
 Return self-value.
 
 astimezone(...)
 tz -> convert to local time in new timezone tz
 
 ctime(...)
 Return ctime() style string.
 
 date(...)
 Return date object with same year, month and day.
 
 dst(...)
 Return self.tzinfo.dst(self).
 
 isoformat(...)
 [sep] -> string in ISO 8601 format, YYYY-MM-DDT[HH[:MM[:SS[.mmm[uuu]]]]][+HH:MM].
 sep is used to separate the year from the time, and defaults to 'T'.
 timespec specifies what components of the time to include (allowed values are 'auto', 'hours', 'minutes', 'seconds', 'milliseconds', and 'microseconds').
 
 replace(...)
 Return datetime with new specified fields.
 
 time(...)
 Return time object with same time but with tzinfo=None.
 
 timestamp(...)
 Return POSIX timestamp as float.
 
 timetuple(...)
 Return time tuple, compatible with time.localtime().
 
 timetz(...)
 Return time object with same time and tzinfo.
 
 tzname(...)
 Return self.tzinfo.tzname(self).
 
 utcoffset(...)
 Return self.tzinfo.utcoffset(self).
 
 utctimetuple(...)
 Return UTC time tuple, compatible with time.localtime().
 
 ----------------------------------------------------------------------
 Class methods defined here:
 
 combine(...) from builtins.type
 date, time -> datetime with same date and time fields
 
 fromisoformat(...) from builtins.type
 string -> datetime from datetime.isoformat() output
 
 fromtimestamp(...) from builtins.type
 timestamp[, tz] -> tz's local time from POSIX timestamp.
 
 now(tz=None) from builtins.type
 Returns new datetime object representing current time local to tz.
 
 tz
 Timezone object.
 
 If no tz is specified, uses local timezone.
 
 strptime(...) from builtins.type
 string, format -> new datetime parsed from a string (like time.strptime()).
 
 utcfromtimestamp(...) from builtins.type
 Construct a naive UTC datetime from a POSIX timestamp.
 
 utcnow(...) from builtins.type
 Return a new datetime representing UTC day and time.
 
 ----------------------------------------------------------------------
 Static methods defined here:
 
 __new__(*args, **kwargs) from builtins.type
 Create and return a new object. See help(type) for accurate signature.
 
 ----------------------------------------------------------------------
 Data descriptors defined here:
 
 fold
 
 hour
 
 microsecond
 
 minute
 
 second
 
 tzinfo
 
 ----------------------------------------------------------------------
 Data and other attributes defined here:
 
 max = datetime.datetime(9999, 12, 31, 23, 59, 59, 999999)
 
 min = datetime.datetime(1, 1, 1, 0, 0)
 
 resolution = datetime.timedelta(microseconds=1)
 
 ----------------------------------------------------------------------
 Methods inherited from date:
 
 __format__(...)
 Formats self with strftime.
 
 isocalendar(...)
 Return a 3-tuple containing ISO year, week number, and weekday.
 
 isoweekday(...)
 Return the day of the week represented by the date.
 Mond

In [0]:
import datetime

today_date = datetime.datetime.now()

print('Date : ',today_date)
print('Year : ',today_date.year)
print('Month : ',today_date.month)
print('Day : ',today_date.day)
print(today_date.strftime("%A"))
print(today_date.strftime("%a"))
print(today_date.strftime("%B"))
print(today_date.strftime("%b"))
print(today_date.strftime("%H"))
print(today_date.strftime("%I"))

Date : 2021-05-05 04:15:56.806032
Year : 2021
Month : 5
Day : 5
Wednesday
Wed
May
May
04
04

In [0]:
help(datetime)

Help on module datetime:

NAME
 datetime - Fast implementation of the datetime type.

CLASSES
 builtins.object
 date
 datetime
 time
 timedelta
 tzinfo
 timezone
 
 class date(builtins.object)
 date(year, month, day) --> date object
 
 Methods defined here:
 
 __add__(self, value, /)
 Return self+value.
 
 __eq__(self, value, /)
 Return self==value.
 
 __format__(...)
 Formats self with strftime.
 
 __ge__(self, value, /)
 Return self>=value.
 
 __getattribute__(self, name, /)
 Return getattr(self, name).
 
 __gt__(self, value, /)
 Return self>value.
 
 __hash__(self, /)
 Return hash(self).
 
 __le__(self, value, /)
 Return self<=value.
 
 __lt__(self, value, /)
 Return self<value.
 
 __ne__(self, value, /)
 Return self!=value.
 
 __radd__(self, value, /)
 Return value+self.
 
 __reduce__(...)
 __reduce__() -> (cls, state)
 
 __repr__(self, /)
 Return repr(self).
 
 __rsub__(self, value, /)
 Return value-self.
 
 __str__(self, /)
 Return str(self).
 
 __sub__(self, value, /)
 Return self-value.
 
 ctime(...)
 Return ctime() style string.
 
 isocalendar(...)
 Return a 3-tuple containing ISO year, week number, and weekday.
 
 isoformat(...)
 Return string in ISO 8601 format, YYYY-MM-DD.
 
 isoweekday(...)
 Return the day of the week represented by the date.
 Monday == 1 ... Sunday == 7
 
 replace(...)
 Return date with new specified fields.
 
 strftime(...)
 format -> strftime() style string.
 
 timetuple(...)
 Return time tuple, compatible with time.localtime().
 
 toordinal(...)
 Return proleptic Gregorian ordinal. January 1 of year 1 is day 1.
 
 weekday(...)
 Return the day of the week represented by the date.
 Monday == 0 ... Sunday == 6
 
 ----------------------------------------------------------------------
 Class methods defined here:
 
 fromisoformat(...) from builtins.type
 str -> Construct a date from the output of date.isoformat()
 
 fromordinal(...) from builtins.type
 int -> date corresponding to a proleptic Gregorian ordinal.
 
 fromtimestamp(...) from builtins.type
 timestamp -> local date from a POSIX timestamp (like time.time()).
 
 today(...) from builtins.type
 Current date or datetime: same as self.__class__.fromtimestamp(time.time()).
 
 ----------------------------------------------------------------------
 Static methods defined here:
 
 __new__(*args, **kwargs) from builtins.type
 Create and return a new object. See help(type) for accurate signature.
 
 ----------------------------------------------------------------------
 Data descriptors defined here:
 
 day
 
 month
 
 year
 
 ----------------------------------------------------------------------
 Data and other attributes defined here:
 
 max = datetime.date(9999, 12, 31)
 
 min = datetime.date(1, 1, 1)
 
 resolution = datetime.timedelta(days=1)
 
 class datetime(date)
 datetime(year, month, day[, hour[, minute[, second[, microsecond[,tzinfo]]]]])
 
 The year, month and day arguments are required. tzinfo may be None, or an
 instance of a tzinfo subclass. The remaining arguments may be ints.
 
 Method resolution order:
 datetime
 date
 builtins.object
 
 Methods defined here:
 
 __add__(self, value, /)
 Return self+value.
 
 __eq__(self, value, /)
 Return self==value.
 
 __ge__(self, value, /)
 Return self>=value.
 
 __getattribute__(self, name, /)
 Return getattr(self, name).
 
 __gt__(self, value, /)
 Return self>value.
 
 __hash__(self, /)
 Return hash(self).
 
 __le__(self, value, /)
 Return self<=value.
 
 __lt__(self, value, /)
 Return self<value.
 
 __ne__(self, value, /)
 Return self!=value.
 
 __radd__(self, value, /)
 Return value+self.
 
 __reduce__(...)
 __reduce__() -> (cls, state)
 
 __reduce_ex__(...)
 __reduce_ex__(proto) -> (cls, state)
 
 __repr__(self, /)
 Return repr(self).
 
 __rsub__(self, value, /)
 Return value-self.
 
 __str__(self, /)
 Return str(self).
 
 __sub__(self, value, /)
 Return self-value.
 
 astimezone(...)
 tz -> convert to local time in new timezone tz
 
 ctime(...)
 Return ctime() style string.
 
 date(...)
 R

In [0]:
import datetime

x = datetime.datetime(2020, 5, 17, 23, 23, 10)

print(x)

2020-05-17 23:23:10


In [0]:
import time;      # This is required to include time module.

localtime = time.localtime(time.time())
print ("Local current time is :", localtime)
type(localtime)

### Data Type Conversion

* str() : Used to convert integer into a string.
* int(a,base) : This function converts any data type to integer. ‘Base’ specifies the base in which string is if data type is string.
* float() : This function is used to convert any data type to a floating point number
* ord() : This function is used to convert a character to integer.
* hex() : This function is to convert integer to hexadecimal string.
* oct() : This function is to convert integer to octal string.
* tuple() : This function is used to convert to a tuple.
* set() : This function returns the type after converting to set.
* list() : This function is used to convert any data type to a list type.
* dict() : This function is used to convert a tuple of order (key,value) into a dictionary.
* complex(real,imag) : : This function converts real numbers to complex(real,imag) number.
* chr(number) : : This function converts number to its corresponding ASCII character.





In [0]:
complex_num = 8+4j
complex_num1 =4+3j
print(complex_num-complex_num1)

(4+1j)

#### Convert String To Bytes


In [0]:
string = "This is sample String."
# string with encoding 'utf-8'
bytes = bytes(string, 'utf-8')
print(bytes)

#### Convert int to bytes ..

In [0]:
#int
#str

In [0]:
a=10
b='88'
print(a+int(b))

In [0]:
a=10
b='88'
print(a+int(b))

In [0]:
c=str(a)+b
print(c)

In [0]:
#create a variable with integer value.
a=100
print("The type of variable having value", a, " is ", type(a))
#create a variable with float value.
b=10.2345
print("The type of variable having value", b, " is ", type(b))
#create a variable with complex value. (real part + img part)
c=100+3j
print("The type of variable having value", c, " is ", type(c))

The type of variable having value 100 is <class 'int'>
The type of variable having value 10.2345 is <class 'float'>
The type of variable having value (100+3j) is <class 'complex'>

In [0]:
v1 = int(2.7) # 2
v2 = int(-3.9) # -3
v3 = int("2") # 2
v4 = int("11", 16) # 17, base 16
v6 = float(2) # 2.0
v7 = float("2.7") # 2.7
v8 = float("2.7E-2") # 0.027
v9 = float(False) # 0.0
vA = float(True) # 1.0
vB = str(4.5) # "4.5"
vC = str([1, 3, 5]) # "[1, 3, 5]"
vD = bool(0) # False; bool fn since Python 2.2.1
vE = bool(3) # True
vF = bool([]) # False - empty list
vG = bool([False]) # True - non-empty list
vH = bool({}) # False - empty dict; same for empty tuple
vI = bool("") # False - empty string
vJ = bool(" ") # True - non-empty string
vK = bool(None) # False
vL = bool(len) # True
vM = set([1, 2])
vN = list(vM)
vO = list({1: "a", 2: "b"}) # dict -> list of keys
vP = tuple(vN)
vQ = list("abc") # ['a', 'b', 'c']


#### sys.getsizeof()
* Return the size of an object in bytes. The object can be any type of object. 
* All built-in objects will return correct results, but this does not have to hold true for third-party extensions as it is implementation specific.

In [0]:
import sys
a='this is no data'
sys.getsizeof(a)

Out[1]: 64

##### Program to print ASCII Value of a character
* ord() : It coverts the given string of length one, return an integer representing the unicode code point of the character. 
* For example, ord(‘a’) returns the integer 97.

In [0]:
char='!'
print("The ASCII value of ",char," is : ",ord(char))
char='*'
print("The ASCII value of ",char," is : ",ord(char))
char='%'
print("The ASCII value of ",char," is : ",ord(char))
char='$'
print("The ASCII value of ",char," is : ",ord(char))
char='&'
print("The ASCII value of ",char," is : ",ord(char))
char='-'
print("The ASCII value of ",char," is : ",ord(char))
char='='
print("The ASCII value of ",char," is : ",ord(char))

The ASCII value of ! is : 33
The ASCII value of * is : 42
The ASCII value of % is : 37
The ASCII value of $ is : 36
The ASCII value of & is : 38
The ASCII value of - is : 45
The ASCII value of = is : 61